### ISSUE TITLE
Widget Time Window Callback

### ISSUE THREAD

**Issue by vislink** [thingsboard/thingsboard#11408]
**Component**

 * Generic

**Description**
Hi,

How does one set a callback function that gets executed when a user changes the widget time window (for e.g., switches from realtime to history, or simply changes the time window size)?

I'm developing a custom time series widget that uses Chart.js ( following the  example here:  https://thingsboard.io/docs/user-guide/contribution/widgets-development/#time-series-example). 
 
I need to execute some  code when the above event is  triggered but can't figure out how to set this hook.

There isn't much documentation available on WidgetSubscriptionCallbacks in the Thingsboard docs. I'm fairly new to TB, so any pointers would be appreciated. 

Thanks.

**Environment**

 * OS:  Ubuntu
 * ThingsBoard: 3.7.0
 * Browser: Chrome

**Comment by devaskim** [comment]
Try this
```
self.ctx.dashboard.dashboardTimewindowChanged.subscribe(
    newTimewindows => {
        console.log('Timewindows Updated);
    }
);
```

**Comment by vislink** [comment]
Thanks @devaskim. That works if the main dashboard time window gets updated. But along those lines,  I was able to place the following hook in `self.onInit()`


```
self.widgetTimewindowChangedSubscription = self.ctx.defaultSubscription.widgetTimewindowChanged.subscribe(
        (timewindow) => {
          some_function()
        }
    );

```
(based on what I found [here:](https://github.com/thingsboard/thingsboard/blob/9e54196ce8f7afcde6a28037f10419be8065f061/ui-ngx/src/app/modules/home/components/widget/lib/timeseries-table-widget.component.ts#L243))
The above only gets called when the widget's time window is updated, which is what I needed. 
Thank you for the hint.
